stdlib/std.jsonnet

```diff
@@ -1041,7 +1042,43 @@ limitations under the License.
         std.join('', lines);
     aux(value, [], ''),
 
-  manifestYamlDoc(value, indent_array_in_object=false)::
+  manifestYamlDoc(value, indent_array_in_object=false, quote_keys=true)::
+    local isReserved(key) =
+      local reserved = ['true', 'false', 'null'];
+      local bad = [word for word in reserved if word == key];
+      if std.length(bad) > 0 then
+        true
+      else false;
+    local bareSafe(key) =
```

Comment:
Here we would need a lengthy comment explaining the approach to determining .

One way to approach that is first lay down the conditions for it being a valid value (as opposed to other syntax elements).

Then we need to consider all the value types it can resolve to and explain what are the sufficient conditions to check for each to make sure it's not classified as anything other than a string.

There should also be an explanation that the considered types (tags in yaml parlance) are sufficent (i.e. that there are no othere are no other types that we haven't considered). 

All of this should be substantiated by the appropriate links to the yaml documentation or other resources.

Reply:
There is now comprehensive details here for each step of the validation procedure, along with links to the YAML spec.

I included a bit that mentions the remaining types not explicitly called out here all require characters considered invalid for potential output without quotes.